File: ProxyChanger-VB/Form1.vb
Imports Microsoft.Win32

Namespace ProxyChanger
    Public Partial Class Form1
        Inherits Windows.Forms.Form
        Private reg As RegistryKey
        Public Shared path As String = "Software\Microsoft\Windows\CurrentVersion\Internet Settings"
        Public Shared x As Integer
        Public Shared y As Integer

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub btnStart_Click(ByVal sender As Object, ByVal e As EventArgs)
            btnStart.Enabled = False
            btnStop.Enabled = True
            txtProxy.Enabled = False
            reg = Registry.CurrentUser.OpenSubKey(path, True)
            Dim proxy = txtProxy.Text
            reg.SetValue("ProxyEnable", 1)
            reg.SetValue("ProxyServer", proxy)
            pnlStatus.BackColor = Drawing.Color.Green
            lblProxyStatus.Text = proxy
            lblProxyStatus.ForeColor = Drawing.Color.White
            x = (pnlStatus.Size.Width - lblProxyStatus.Size.Width) / 2
            y = (pnlStatus.Size.Height - lblProxyStatus.Size.Height) / 2
            lblProxyStatus.Location = New Drawing.Point(x, y)
        End Sub

        Private Sub btnStop_Click(ByVal sender As Object, ByVal e As EventArgs)
            btnStart.Enabled = True
            btnStop.Enabled = False
            txtProxy.Enabled = True
            reg.SetValue("ProxyEnable", 0)
            pnlStatus.BackColor = Drawing.Color.Red
            lblProxyStatus.ForeColor = Drawing.Color.White
            lblProxyStatus.Text = "No proxy"
            x = (pnlStatus.Size.Width - lblProxyStatus.Size.Width) / 2
            y = (pnlStatus.Size.Height - lblProxyStatus.Size.Height) / 2
            lblProxyStatus.Location = New Drawing.Point(x, y)
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            x = (pnlStatus.Size.Width - lblProxyStatus.Size.Width) / 2
            y = (pnlStatus.Size.Height - lblProxyStatus.Size.Height) / 2
            lblProxyStatus.Location = New Drawing.Point(x, y)

        End Sub
    End Class
End Namespace

File: ProxyChanger-VB/Form1.cs
using Microsoft.Win32;
using System.Runtime.InteropServices;

namespace ProxyChanger
{
    public partial class Form1 : Form
    {       
        RegistryKey reg;
        public static string path = @"Software\Microsoft\Windows\CurrentVersion\Internet Settings";
        public static int x;
        public static int y;

        public Form1()
        {
            InitializeComponent();
        }

        private void btnStart_Click(object sender, EventArgs e)
        {
            btnStart.Enabled = false;
            btnStop.Enabled = true;
            txtProxy.Enabled = false;
            reg = Registry.CurrentUser.OpenSubKey(path, true);
            string proxy = txtProxy.Text;
            reg.SetValue("ProxyEnable", 1);
            reg.SetValue("ProxyServer", proxy);
            pnlStatus.BackColor = Color.Green;
            lblProxyStatus.Text = proxy;
            lblProxyStatus.ForeColor = Color.White;
            x = (pnlStatus.Size.Width - lblProxyStatus.Size.Width) / 2;
            y = (pnlStatus.Size.Height - lblProxyStatus.Size.Height) / 2;
            lblProxyStatus.Location = new Point(x, y);
        }

        private void btnStop_Click(object sender, EventArgs e)
        {
            btnStart.Enabled = true;
            btnStop.Enabled = false;
            txtProxy.Enabled = true;
            reg.SetValue("ProxyEnable", 0);           
            pnlStatus.BackColor = Color.Red;
            lblProxyStatus.ForeColor = Color.White;
            lblProxyStatus.Text = "No proxy";
            x = (pnlStatus.Size.Width - lblProxyStatus.Size.Width) / 2;
            y = (pnlStatus.Size.Height - lblProxyStatus.Size.Height) / 2;
            lblProxyStatus.Location = new Point(x, y);
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            x = (pnlStatus.Size.Width - lblProxyStatus.Size.Width) / 2;
            y = (pnlStatus.Size.Height - lblProxyStatus.Size.Height) / 2;
            lblProxyStatus.Location = new Point(x, y);
            
        }
    }
}